docs/t8ntool-ref.rst

```diff
@@ -0,0 +1,298 @@
+            {
+               "address": "0x0000000000000000000000000000000000000aaaa",
+               "storageKeys": [
+                  "0x0000000000000000000000000000000000000000000000000000000000000000",
+                  "0x0000000000000000000000000000000000000000000000000000000000000012"
+               ]
+            },
+            {
+               "address": "0x0000000000000000000000000000000000000aaab",
+               "storageKeys": [
+                  "0x00000000000000000000000000000000000000000000000000000000000060A7",
+                  "0x0000000000000000000000000000000000000000000000000000000000000012"
+               ]
+            }
+         ]
+         "v": "0x0",
+         "r": "0x0",
+         "s": "0x0",
+         "secretKey": "0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"
+      }
+   ]
+
+
+Environment File
+----------------
+This file is a map with the execution environment. This is an input to the tool, 
+typically called `env.json`. It has these fields:
+
+
+* `currentCoinbase`
+* `currentDifficulty`
+* `currentGasLimit`
+* `currentNumber`
+* `currentTimestamp`
+* `previousHash`, the hash of the previous (`currentNumber`-1) block
+* `blockHashes`, a map of historical block numbers and their hashes
+
+.. note::
+
+   Some tests include multiple blocks. In that case, the test software runs 
+   `t8ntool` multiple times, one per block. 
+
+
+Example
+^^^^^^^
+
+::
+
+  {
+     "currentCoinbase" : "0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba",
+     "currentDifficulty" : "0x020000",
+     "currentGasLimit" : "0x05f5e100",
+     "currentNumber" : "0x01",
+     "currentTimestamp" : "0x03e8",
+     "previousHash" : "0xe729de3fec21e30bea3d56adb01ed14bc107273c2775f9355afb10f594a10d9e",
+     "blockHashes" : {
+         "0" : "0xe729de3fec21e30bea3d56adb01ed14bc107273c2775f9355afb10f594a10d9e"
+     }
+  }
+
+
+Allocation Files
+----------------
+These files show the state of various accounts and contracts on the blockchain.
+There are two of them, `alloc.json` which is the input state and `outAlloc.json`
+which is the output state.
+
+The file is a map of `address` values to account information. The account 
+information that can be provided is:
+
+* `balance` 
+* `code` (in machine language format)
+* `nonce`
+* `storage`
+
+Example
+^^^^^^^
+
+::
+
+   {
+       "a94f5374fce5edbc8e2a8697c15331677e6ebf0b": {
+           "balance": "0x5ffd4878be161d74",
+           "code": "0x5854505854",
+           "nonce": "0xac",
+           "storage": {
+              "0x0000000000000000000000000000000000000000000000000000000000000000": 
+              "0x0000000000000000000000000000000000000000000000000000000000000004"
+           }
+        },
+        "0x8a8eafb1cf62bfbeb1741769dae1a9dd47996192":{
+           "balance": "0xfeedbead",
+           "nonce" : "0x00"
+        }
+   }
+
+
+
+
+
+
+Result File
+-----------
+This file is typically called `out.json`. It is the post state after 
```

Comment:
The 'typically called' is an implementation-specific choice. It's up to the implementation to decide what to do if the user didn't specify a filename or `stdout`

Reply:
Having the filenames would be helpful, so I'm changing it to:

This is an input to the tool, which `retesteth` calls `txs.json`.

OK?